File: Easy/27. Remove Element/ans.js
var removeElement = function(nums, val) {
    let newLen = 0;
    for (let i=0; i<nums.length; i++) {
        if (nums[i] == val) {
            continue;
        }
        nums[newLen] = nums[i]; 
        newLen++;
    }
    return newLen;
};

// When nums[j] equals to the given value, skip this element by incrementing j. 
// As long as nums[j] != val, we copy nums[j] to nums[i] 
// and increment both indexes at the same time.Repeat the process 
// until j reaches the end of the array and the new length is i.

File: Easy/26. Remove Duplicates from Sorted Array/ans.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:    
        length = 0
        if len(nums) == 0: 
            return length
        for i in range(1,len(nums)):
            if nums[length] < nums[i]:
                length += 1
                nums[length] = nums[i]
        return length+1

# Since the array is already sorted, we can keep two pointers i and j, 
# where i is the slow-runner while j is the fast-runner. 
# As long as nums[i] = nums[j]nums[i]=nums[j], 
# we increment jj to skip the duplicate.

# When we encounter nums[j] not equal to nums[i], 
# the duplicate run has ended so we must copy its value to nums[i + 1]. 
# i is then incremented and we repeat the same process again 
# until j reaches the end of array.

